crates/evm/core/src/fork/backend.rs

```diff
@@ -603,12 +622,12 @@ impl SharedBackend {
         let (backend, backend_rx) = channel(1);
         let cache = Arc::new(FlushJsonBlockCacheDB(Arc::clone(db.cache())));
         let handler = BackendHandler::new(provider, db, backend_rx, pin_block);
-        (Self { backend, cache, data_accesses, chain, state_lookup }, handler)
+        (Self { backend, cache, data_accesses, chain, state_lookup, code_cache }, handler)
     }
 
     /// Updates the pinned block to fetch data from
-    pub fn set_pinned_block(&self, block: impl Into<BlockId>) -> eyre::Result<()> {
-        let req = BackendRequest::SetPinnedBlock(block.into());
```

Comment:
Why are we making this change?

Reply:
A BlockId can be pinned to a tag or a block number. 

The BackendHandler needs to be pinned to a specific block number to use the code cache.

If we used the BlockId with the latest tag in place of a block number then we wouldn't be tracking block numbers properly and instead would be overwriting one entry as the chain progresses.